app/src/componentLibrary/popover/withPopover/withPopover.jsx

```diff
@@ -14,19 +14,35 @@
  * limitations under the License.
  */
 
-import React, { useRef, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import classNames from 'classnames/bind';
+import { activeModalSelector, hideModalAction } from 'controllers/modal';
+import { useDispatch, useSelector } from 'react-redux';
 import { Popover } from '../popover';
 import styles from './withPopover.scss';
 
 const cx = classNames.bind(styles);
 
-export const withPopover = ({ ContentComponent, popoverWrapperClassName, ...popoverConfig }) => (
-  WrappedComponent,
-) => (props) => {
+export const withPopover = ({
+  ContentComponent,
+  popoverWrapperClassName,
+  popoverId,
+  ...popoverConfig
+}) => (WrappedComponent) => (props) => {
   const parentRef = useRef();
+  const dispatch = useDispatch();
+  const activePopover = useSelector(activeModalSelector)?.id || '';
```

Comment:
Please avoid using app specifics in the common components, so move it upper and manage in the parent component.
I suppose it can be achieved by managing `isOpened` in the parent component and providing `onClose` callback.

Reply:
Done.